src/main/scala/Core/CSR.scala

```diff
@@ -69,12 +71,15 @@ class CSR(implicit val conf: FlexpretConfiguration) extends Module {
   val reg_tmodes = RegInit(VecInit(conf.initialTmodes.toSeq))
   // exception handling
   val reg_evecs = Reg(Vec(conf.threads, UInt()))
-  val reg_epcs = Reg(Vec(conf.threads, UInt())) // RO?
+  val reg_mepcs = Reg(Vec(conf.threads, UInt())) // RO?
+
   val reg_causes = Reg(Vec(conf.threads, UInt())) // RO
   val reg_sup0 = Reg(Vec(conf.threads, UInt()))
   // timing instructions
   val reg_time = RegInit(0.U(64.W)) // RO, less bits if !conf.stats
-  val reg_compare = Reg(Vec(conf.threads, UInt(conf.timeBits.W)))
+  val reg_compare_du_wu = Reg(Vec(conf.threads, UInt(conf.timeBits.W)))
+  val reg_compare_ie_ee = Reg(Vec(conf.threads, UInt(conf.timeBits.W)))
```

Comment:
Need one register for DU/WU and one for IE/EE, otherwise they will not work together